--- origsrc/telepathy-glib-0.10.0/telepathy-glib/Makefile.am	2010-01-25 20:21:04.721646500 -0600
+++ src/telepathy-glib-0.10.0/telepathy-glib/Makefile.am	2010-01-25 20:21:34.604355700 -0600
@@ -76,6 +76,7 @@ noinst_LTLIBRARIES = libtelepathy-glib-i
 # that vim quickfix mode (:make) doesn't interpret the libtool --mode=link
 # command as an error message in a bizarrely named file
 libtelepathy_glib_la_LDFLAGS = \
+    -no-undefined \
     -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
 libtelepathy_glib_la_LIBADD = libtelepathy-glib-internal.la
 nodist_libtelepathy_glib_la_SOURCES =
