--- origsrc/telepathy-glib-0.12.6/telepathy-glib/abi.am	2010-12-10 10:23:56.000000000 -0600
+++ src/telepathy-glib-0.12.6/telepathy-glib/abi.am	2010-12-15 21:25:06.228278700 -0600
@@ -55,7 +55,7 @@ ABI_LISTS = \
 # that vim quickfix mode (:make) doesn't interpret the libtool --mode=link
 # command as an error message in a bizarrely named file
 libtelepathy_glib_la_LDFLAGS = \
-    -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
+    -no-undefined -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
 libtelepathy_glib_la_LIBADD = libtelepathy-glib-internal.la
 nodist_libtelepathy_glib_la_SOURCES =
 libtelepathy_glib_la_DEPENDENCIES = libtelepathy-glib-internal.la abi.am
--- origsrc/telepathy-glib-0.12.6/tests/Makefile.am	2010-12-10 10:23:56.000000000 -0600
+++ src/telepathy-glib-0.12.6/tests/Makefile.am	2010-12-15 22:44:27.394438700 -0600
@@ -56,8 +56,8 @@ test_asv_SOURCES = \
 test_capabilities_SOURCES = \
     capabilities.c
 test_capabilities_LDADD = \
-    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(top_builddir)/telepathy-glib/libtelepathy-glib-internal.la \
+    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(GLIB_LIBS)
 
 test_heap_SOURCES = \
