--- origsrc/telepathy-glib-0.14.5/telepathy-glib/abi.am
+++ src/telepathy-glib-0.14.5/telepathy-glib/abi.am
@@ -77,6 +77,7 @@ ABI_LISTS = \
 # command as an error message in a bizarrely named file
 libtelepathy_glib_la_LDFLAGS = \
     $(AM_LDFLAGS) \
+    -no-undefined \
     -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
 libtelepathy_glib_la_LIBADD = libtelepathy-glib-internal.la
 nodist_libtelepathy_glib_la_SOURCES =
--- origsrc/telepathy-glib-0.12.6/tests/Makefile.am	2010-12-10 10:23:56.000000000 -0600
+++ src/telepathy-glib-0.12.6/tests/Makefile.am	2010-12-15 22:44:27.394438700 -0600
@@ -56,8 +56,8 @@ test_asv_SOURCES = \
 test_capabilities_SOURCES = \
     capabilities.c
 test_capabilities_LDADD = \
-    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(top_builddir)/telepathy-glib/libtelepathy-glib-internal.la \
+    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(GLIB_LIBS)
 
 test_heap_SOURCES = \
@@ -67,8 +67,8 @@ test_capabilities_LDADD = \
 test_contact_search_result_SOURCES = \
     contact-search-result.c
 test_contact_search_result_LDADD = \
-    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(top_builddir)/telepathy-glib/libtelepathy-glib-internal.la \
+    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(GLIB_LIBS)
 
 test_dtmf_player_SOURCES = dtmf-player.c
--- origsrc/telepathy-glib-0.14.5/tests/dbus/Makefile.am	2011-04-20 09:01:32.000000000 -0500
+++ src/telepathy-glib-0.14.5/tests/dbus/Makefile.am	2011-05-09 01:56:23.599405200 -0500
@@ -198,8 +198,8 @@ test_client_channel_factory_SOURCES = cl
 test_cm_message_SOURCES = \
     cm-message.c
 test_cm_message_LDADD = \
-    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(top_builddir)/telepathy-glib/libtelepathy-glib-internal.la \
+    $(top_builddir)/tests/lib/libtp-glib-tests.la \
     $(GLIB_LIBS)
 
 test_text_channel_SOURCES = text-channel.c
