--- origsrc/telepathy-glib-0.7.14/telepathy-glib/Makefile.am	2008-08-24 03:42:35.193162800 -0500
+++ src/telepathy-glib-0.7.14/telepathy-glib/Makefile.am	2008-08-24 03:48:15.567724600 -0500
@@ -81,7 +81,7 @@
 	diff -c _gen/versioned-abi.txt _gen/abi.txt
 
 _gen/abi.txt: libtelepathy-glib-internal.la Makefile.am
-	$(NM) .libs/libtelepathy-glib-internal.a > _gen/abi.nm
+	$(NM) -C .libs/libtelepathy-glib-internal.a > _gen/abi.nm
 	grep " [DT] " < _gen/abi.nm > _gen/abi.funcs
 	cut -d" " -f3 < _gen/abi.funcs > _gen/abi.funcnames
 	grep "^tp" < _gen/abi.funcnames > _gen/abi.tpfuncnames
