--- origsrc/telepathy-glib-0.7.10/telepathy-glib/Makefile.am	2008-06-06 08:55:46.000000000 -0500
+++ src/telepathy-glib-0.7.10/telepathy-glib/Makefile.am	2008-06-24 12:25:10.796875000 -0500
@@ -42,6 +42,7 @@
 # that vim quickfix mode (:make) doesn't interpret the libtool --mode=link
 # command as an error message in a bizarrely named file
 libtelepathy_glib_la_LDFLAGS = \
+    -no-undefined \
     -version-info "$(LT_CURRENT)":"$(LT_REVISION)":"$(LT_AGE)"
 libtelepathy_glib_la_LIBADD = libtelepathy-glib-internal.la
 libtelepathy_glib_la_SOURCES =
--- origsrc/telepathy-glib-0.7.10/tests/lib/Makefile.am	2008-05-07 11:05:09.000000000 -0500
+++ src/telepathy-glib-0.7.10/tests/lib/Makefile.am	2008-06-24 14:03:18.046875000 -0500
@@ -11,6 +11,11 @@
     textchan-null.c \
     textchan-null.h
 
+libtp_glib_tests_la_LIBADD = \
+    $(DBUS_LIBS) \
+    $(GLIB_LIBS) \
+    $(TP_GLIB_LIBS)
+
 check_c_sources = *.c
 include $(top_srcdir)/tools/check-coding-style.mk
 check-local: check-coding-style
--- origsrc/telepathy-glib-0.7.20/tests/dbus/Makefile.am	2008-12-14 11:36:28.000000000 -0600
+++ src/telepathy-glib-0.7.20/tests/dbus/Makefile.am	2008-12-15 22:21:58.134500000 -0600
@@ -22,8 +22,8 @@
 TESTS = $(noinst_PROGRAMS)
 
 LDADD = \
-    $(top_builddir)/telepathy-glib/libtelepathy-glib.la \
-    $(top_builddir)/tests/lib/libtp-glib-tests.la
+    $(top_builddir)/tests/lib/libtp-glib-tests.la \
+    $(top_builddir)/telepathy-glib/libtelepathy-glib.la
 
 test_channel_introspect_SOURCES = channel-introspect.c
 
